src/Controller/ProgrammeController.php

```diff
@@ -69,4 +75,34 @@ public function showExactSearchByName(Request $request): Response
 
         return new JsonResponse($serializedData, Response::HTTP_OK, [], true);
     }
+
+    /**
+     * @Route(path="/sort/participants", methods={"GET"})
+     */
+    public function showSortedByCertainField(Request $request): Response
+    {
+        $sortType = $request->query->get('sort');
+
+        $field = $request->query->get('field');
+
+        $sorted = $this->programmeRepository->getSorted($field, $sortType);
+
+        $serializedData = $this->serializer->serialize($sorted, 'json', ['groups' => 'api:programme:all']);
+
+        return new JsonResponse($serializedData, Response::HTTP_OK, [], true);
+    }
+
+    /**
+     * @Route(path="/paginate", methods={"GET"})
+     */
+    public function showPaginated(Request $request): Response
```

Comment:
Exposing this method has the following issues:
- your API is no longer RESTful
- pagination should be possible on the main GET route for programs (through an optional query parameter)